src/adagrams.js

```diff
@@ -1,15 +1,123 @@
+const letterPOOL = {
+  A: 9,
+  B: 2,
+  C: 2,
+  D: 4,
+  E: 12,
+  F: 2,
+  G: 3,
+  H: 2,
+  I: 9,
+  J: 1,
+  K: 1,
+  L: 4,
+  M: 2,
+  N: 6,
+  O: 8,
+  P: 2,
+  Q: 1,
+  R: 6,
+  S: 4,
+  T: 6,
+  U: 4,
+  V: 2,
+  W: 2,
+  X: 1,
+  Y: 2,
+  Z: 1,
+};
+
+const scoreChart = {
+  A: 1,
+  E: 1,
+  I: 1,
+  O: 1,
+  U: 1,
+  L: 1,
+  N: 1,
+  R: 1,
+  S: 1,
+  T: 1,
+  D: 2,
+  G: 2,
+  B: 3,
+  C: 3,
+  M: 3,
+  P: 3,
+  F: 4,
+  H: 4,
+  V: 4,
+  W: 4,
+  Y: 4,
+  K: 5,
+  J: 8,
+  X: 8,
+  Q: 10,
+  Z: 10,
+};
+
 export const drawLetters = () => {
-  // Implement this method for wave 1
+  const arrayLetter = Object.keys(letterPOOL);
+  let letters = [];
+  let letterFreq = {};
+  while (letters.length < 10) {
+    const letter = arrayLetter[Math.floor(Math.random() * arrayLetter.length)];
+    // console.log(letter);
+    if (letter in letterFreq) {
+      letterFreq[letter] += 1;
+    } else letterFreq[letter] = 1;
+    if (letterFreq[letter] <= letterPOOL[letter]) {
+      letters.push(letter);
+    }
+  }
+  return letters;
```

Comment:
Overall I see what you were trying to achieve here, just a few small adjustments!

1) Your function currently forces a false randomness. It will randomly pick one of the 26 letters and remove one,
but it ignores the fact that we have different letters with different frequencies. To get around this, the main
suggestion would be to make a global dictionary like the one you had, and then write a helper function that
converts that dictionary into a list or string that holds every single letter possible. Then we can use that helper
function to create our collection of letters as opposed to making a deep copy! This will also make it so that you don't have to build a frequency map to make those comparisons!

2) If we have that collection of all the letters, it makes it really easy to remove letters as you use them!